libraries/AC_AttitudeControl/AC_AttitudeControl.cpp

```diff
@@ -188,15 +188,16 @@ float AC_AttitudeControl::get_slew_yaw_max_degs() const
 // Ensure attitude controller have zero errors to relax rate controller output
 void AC_AttitudeControl::relax_attitude_controllers()
 {
+    // take a copy of the last gyro used by the rate controller before using it
+    Vector3f gyro = _rate_gyro;
```

Comment:
Doesn't this mean we relax to the last loops gyro not this loops? Plane at least calls run last.

Reply:
With the rate thread these things are going on concurrently so there is no first/last - so in fact the opposite is true - a much more up-to-date value is being used. In he non-rate case you might be right - I'll have to have a think about how this might be fixed. Obvious answer is to have two functions. Where is this called from?

Reply:
I meant all of these functions that use `_rate_gyro`. Copter runs the rate controller the first thing in the new loop, so on copter everything that comes later is using the correct value. Plane does all the inputs and then runs the rate controller.

We almost need a `set gyro` call, so both plane and copter do that first with the AHRS update and then whatever comes after will always be using a consistent value.

Reply:
In copter this behaviour is unchanged - the AHRS is updated after the rate loops have run, so in this case it is a more accurate view. Note that _rate_gyro is the current gyro from the sensor here, not the delayed AHRS version - so again a more accurate version.

Reply:
> In copter this behaviour is unchanged

We need to check plane too.

Reply:
I have added a change which means plane now gets the latest AHRS value for all of these, so NFC. Longer term plane should be doing it the way copter does to minimize latency